### PR TITLE
feat: native transfer handler

### PR DESCRIPTION
**Pull request by Jonathansumner**
Addressing #8: 
- Implements JSON `coin` type within the schema
- Introduces a new handler `handleNativeTransfer`
- Includes schema updates for the `NativeTransferMsg` type and the `NativeTransfer` db entity

### CODE REVIEW

File: src/mappings/mappingHandlers.ts [hunk 1]
```diff
@@ -9,10 +9,17 @@ import {
...
+  NativeTransfer
 } from "../types";
 import {CosmosBlock, CosmosEvent, CosmosMessage, CosmosTransaction,} from "@subql/types-cosmos";
-import {ExecuteContractMsg, DistDelegatorClaimMsg, GovProposalVoteMsg, LegacyBridgeSwapMsg} from "./types";
+import {
+    ExecuteContractMsg,
+    DistDelegatorClaimMsg,
+    GovProposalVoteMsg,
+    LegacyBridgeSwapMsg,
+    NativeTransferMsg
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:21]: This should be in the first import statement

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]
Thanks @Jonathansumner, this looks great!

We just need to clean up that stray import and then I think this is good to go. :smile:

File: src/mappings/mappingHandlers.ts (hunk 1 above)
Comment (Jonathansumner) [reply src/mappings/mappingHandlers.ts]: Got it!

**Review by bryanchriswhite — APPROVED** [review]
Amazing, thanks @Jonathansumner! :raised_hands:

File: src/mappings/mappingHandlers.ts [hunk 2]
```diff
@@ -65,6 +72,20 @@ export async function handleTransaction(tx: CosmosTransaction): Promise<void> {
   await txEntity.save();
 }
 
+export async function handleNativeTransfer(msg: CosmosMessage<NativeTransferMsg>): Promise<void> {
+  logger.info(`[handleMessage] (tx ${msg.tx.hash}): indexing message ${msg.idx + 1} / ${msg.tx.decodedTx.body.messages.length}`)
+  logger.debug(`[handleMessage] (msg.msg): ${JSON.stringify(msg.msg, null, 2)}`)
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:77]: Oh, quick - last minute thing:
```suggestion
  logger.info(`[handleNativeTransfer] (tx ${msg.tx.hash}): indexing message ${msg.idx + 1} / ${msg.tx.decodedTx.body.messages.length}`)
  logger.debug(`[handleNativeTransfer] (msg.msg): ${JSON.stringify(msg.msg, null, 2)}`)
```

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]

**Review by bryanchriswhite — APPROVED** [review]

**Pull request merged**